packages/graphql/src/classes/subscription/cdc/cdc-api.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Cypher from "@neo4j/cypher-builder";
+import type { Driver } from "neo4j-driver";
+import type { CDCEvent, CDCQueryResponse } from "./cdc-types";
+
+export class CDCApi {
+    private driver: Driver;
+    private token: string = "";
+
+    constructor(driver: Driver) {
+        this.driver = driver;
+    }
+
+    public async queryEvents(): Promise<CDCEvent[]> {
+        if (!this.token) {
+            this.token = await this.fetchCurrentToken();
+        }
+
+        const tokenLiteral = new Cypher.Literal(this.token);
+        const queryProcedure = CDCCypher.query(tokenLiteral).yield("id", "event");
+
+        const events = await this.runProcedure<CDCQueryResponse>(queryProcedure);
+
+        this.updateTokenWithLastEvent(events);
+        return events.map((query) => query.event);
+    }
+
+    private async fetchCurrentToken(): Promise<string> {
+        const currentProcedure = CDCCypher.current();
+
+        const result = await this.runProcedure<{ id: string }>(currentProcedure);
+        return result[0]!.id;
```

Comment:
is this really always an array of length 1 that is not undefined? is there no point in checking this in typescript?

Reply:
It should always be. We could check, although checking would only throw an error at this point